### PR TITLE
Bug 1182464 - added info from failing container

### PR DESCRIPTION
**Pull request by soltysh**
@csrwng could you please have a look, since I'm modifying the expected error behavior slightly, I wanted to get rid of the hack that @smarterclayton didn't like, in favor of passing the error output (which is limited to 1024) up the stack.

### CODE REVIEW

**Comment by csrwng** [comment]
LGTM

File: pkg/sti/build.go [hunk 1]
```diff
@@ -105,6 +104,8 @@ func (b *Builder) Build() (*api.Result, error) {
 		if err := bh.execute(api.Assemble); err != nil {
 			return nil, err
 		}
+	} else if ok && err != nil {
+		return nil, errors.NewAssembleError(bh.Request().Tag, e.Output, err)
```
Comment (mfojtik) [inline pkg/sti/build.go]: this is getting a bit hairy, I would refactor this to use switch or re-organize the if's

File: pkg/sti/errors/errors.go [hunk 2]
```diff
@@ -84,15 +84,26 @@ func NewScriptDownloadError(name api.Script, err error) error {
 
 // NewSaveArtifactsError returns a new error which indicates there was a problem
 // calling save-artifacts script
-func NewSaveArtifactsError(name string, err error) error {
+func NewSaveArtifactsError(name, output string, err error) error {
 	return Error{
-		Message:    fmt.Sprintf("saving artifacts for %s failed", name),
+		Message:    fmt.Sprintf("saving artifacts for %s failed:\n%s", name, output),
```
Comment (mfojtik) [inline pkg/sti/errors/errors.go]: the new line will make grepping this super-difficult ;-)

File: pkg/sti/errors/errors.go [hunk 3]
```diff
@@ -100,7 +111,7 @@ func NewBuildError(name string, err error) error {
 		Message:    fmt.Sprintf("building %s failed", name),
 		Details:    err,
 		ErrorCode:  ErrBuild,
-		Suggestion: "check the assemble script for errors",
+		Suggestion: "check the build output for errors",
```
Comment (mfojtik) [inline pkg/sti/errors/errors.go]: maybe suggest how to do that? (increase log-level?)

File: pkg/sti/errors/errors.go (hunk 2 above)
Comment (soltysh) [inline pkg/sti/errors/errors.go]: Not necessarily and if so we'll change it later :bowtie:

File: pkg/sti/errors/errors.go (hunk 3 above)
Comment (soltysh) [inline pkg/sti/errors/errors.go]: That's already happening, see https://github.com/openshift/source-to-image/blob/master/cmd/sti/main.go#L189-L191

File: pkg/sti/build.go (hunk 1 above)
Comment (soltysh) [inline pkg/sti/build.go]: Fixed with switch type.

**Comment by soltysh** [comment]
@mfojtik addressed your comments, how's that?

**Comment by mfojtik** [comment]
@soltysh LGTM

**Comment by soltysh** [comment]
[merge]

**Pull request merged**